torch/_inductor/compile_fx.py

```diff
@@ -552,7 +552,21 @@ def compile_fx_inner(
     context = torch._guards.TracingContext.try_get()
     if context is not None and context.output_strides is not None:
         assert len(context.output_strides) == 0
-        context.output_strides.extend(compiled_graph.output_strides)
+        shape_env = _shape_env_from_inputs(example_inputs)
```

Comment:
Re https://github.com/pytorch/pytorch/pull/128997#issuecomment-2179114601

I thought this looked kind of suspicious but could not see anything concretely wrong with it

Reply:
What's the right way to retrieve the shape env? In the codecache, I look for it in the TracingContext: https://github.com/pytorch/pytorch/blob/4e03263224af813fbf5e0e745e84c13268c48dc7/torch/_inductor/codecache.py#L793-L801
But I guess that's looking incomplete since most calls in this file are using `detect_fake_mode()`, which does more stuff. Then there's a single call to `_shape_env_from_inputs()` (now a 2nd); the TODO in that method makes it sound like it's strictly for not breaking some unit tests?

Reply:
Pretty silly, the problem was `torch/_guards.py` typed stride as List, but it needs to be a tuple so that result of stride() can be compared to it. Previously, we would hard cast it to tuple but mypy would not error (another L for the mypy) but now that I make it explicit, the type needed to be updated to be tuple again (correctly).